### ISSUE TITLE
Change default colors in draw_segmentation_masks()

### ISSUE THREAD

**Issue by ld-ing** [pytorch/vision#7433]
### 🚀 The feature

The current `utils.draw_segmentation_masks()` uses `_generate_color_palette(num_masks)` by default if the colors of the masks are not specified by the user. `_generate_color_palette(num_masks)` has a behavior that it always give black as the first color. As an example:

```
# https://github.com/pytorch/vision/blob/b403bfc771e0caf31efd06d43860b09004f4ac61/torchvision/utils.py#L533
def _generate_color_palette(num_objects: int):
    palette = torch.tensor([2**25 - 1, 2**15 - 1, 2**21 - 1])
    return [tuple((i * palette) % 255) for i in range(num_objects)]

# generate colors for 1-3 objects
for i in range(1, 4):
    print(_generate_color_palette(i))
```
yields
```
[(tensor(0), tensor(0), tensor(0))]
[(tensor(0), tensor(0), tensor(0)), (tensor(1), tensor(127), tensor(31))]
[(tensor(0), tensor(0), tensor(0)), (tensor(1), tensor(127), tensor(31)), (tensor(2), tensor(254), tensor(62))]
```

The proposal is to change this behavior such that black is never used in the color palette. The reason is sometimes users need to draw the segmentation masks on a pure black background, and the current behavior will hide the first object since it is also black, and it may be hard for the users to notice that an object is missing.

### Motivation, pitch

Here is an example of when I'm trying to draw segmentation masks on a black background. 

An image (from COCO) for reference:
<img width="313" alt="image" src="https://user-images.githubusercontent.com/17507835/225939077-d9932eef-5ab3-4f7a-9a39-17cd65565d83.png">

The default behavior:
```
viz = utils.draw_segmentation_masks(
        torch.zeros(img.shape, dtype=torch.uint8), 
        masks=masks, 
        alpha=1)
```
<img width="314" alt="image" src="https://user-images.githubusercontent.com/17507835/225939467-02ba13e6-52c6-47a7-871b-c9a4ff867bd1.png">

The desired behavior would be something like:
```
viz = utils.draw_segmentation_masks(
        torch.zeros(img.shape, dtype=torch.uint8), 
        masks=masks, 
        alpha=1, 
        colors=[tuple(_) for _ in np.random.randint(100, 256, size=(len(masks), 3)).tolist()])
```
<img width="314" alt="image" src="https://user-images.githubusercontent.com/17507835/225939712-faf2b336-fb00-435e-aa6a-f71dac100bf5.png">

### Additional context

The easiest way to modify this is to iterate from the second element of the current `_generate_color_palette()` function, i.e.,

```
def _generate_color_palette_v2(num_objects: int):
    palette = torch.tensor([2**25 - 1, 2**15 - 1, 2**21 - 1])
    return [tuple((i * palette) % 255) for i in range(1,num_objects+1)]
```
which will give
<img width="313" alt="image" src="https://user-images.githubusercontent.com/17507835/225941164-7eabe480-11e8-4fac-bbfc-44e803790eeb.png">

However, personally, I think the colors are still not visually distinct enough (most colors are green/blue-ish), so I wonder what's the intuition behind the current implementation, and if we can somehow improve it.

I'm happy to make a diff, please comment if you have any thoughts.

**Comment by oke-aditya** [comment]
This is a valid point, and it's been tricky how we handle the colors. Mostly to avoid over flexibility. 
I remember mentioning to @NicolasHug about flexible color palette and fixing 1 color per class assignment, but after few discussions we decided not to go for it.

I suppose we can least allow users to pass list of custom colors. Like what we do for drawing boxes, But yes it is far better to provide very sane defaults that work well out of box for many cases. @ld-ing has valid point here.

cc @NicolasHug what do you think?

**Comment by NicolasHug** [comment]
> I suppose we can least allow users to pass list of custom colors

don't we do that already via the `colors` parameter?

Anyway, there's not BC consideration for the default color palette. So if there's a simple way to have a more heterogeneous default palette,  we can go for it.

**Comment by oke-aditya** [comment]
> don't we do that already via the colors parameter?

Oh yes we do that, apologies for not checking. I got confused with the keypoints API where we have locked to white color :(
